### PR TITLE
[GUILD-1380] - Sign-in-with-Ethereum

### PR DESCRIPTION
**Pull request by cs-balazs**
- Adds an alternative message-generation function `getSiweMessage` which is used when sending a request to the verify endpoint
- More details on the BE PR: https://github.com/guildxyz/guild-backend/pull/1421

### CODE REVIEW

File: src/hooks/useSetKeyPair.ts [hunk 1]
```diff
@@ -7,8 +7,32 @@ import { useEffect } from "react"
...
+  addr,
+  method,
+  nonce: nonceRandom,
+  ts,
+  chainId,
+  hash,
+}: MessageParams) {
+  const nonce = `${hash}${nonceRandom}${method ?? 1}`
+
+  // Indentation is important inside the string, extra insentation would be extra whitespace in the string
```
Comment (BrickheadJohnny) [inline src/hooks/useSetKeyPair.ts:24]: nit: typo 😉

**Review by BrickheadJohnny — APPROVED** [review]
I only found 1 typo, otherwise really nice PR as always! 💯

File: src/hooks/useSetKeyPair.ts (hunk 1 above)
Comment (cs-balazs) [reply src/hooks/useSetKeyPair.ts]: Nice catch, fixed in https://github.com/guildxyz/guild.xyz/pull/1185/commits/eed226e3f0a998c8ca87aa081124cf5ee986ac4d :ayy:

**Comment by Jubbyjazzy** [comment]
Deploying

**Comment by dmgs19** [comment]
hi

**Pull request merged**